homeassistant/components/tessie/media_player.py

```diff
@@ -0,0 +1,108 @@
+"""Media Player platform for Tessie integration."""
+from __future__ import annotations
+
+from homeassistant.components.media_player import (
+    MediaPlayerDeviceClass,
+    MediaPlayerEntity,
+    MediaPlayerState,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import TessieDataUpdateCoordinator
+from .entity import TessieEntity
+
+STATES = {
+    "Playing": MediaPlayerState.PLAYING,
+    "Paused": MediaPlayerState.PAUSED,
+    "Stopped": MediaPlayerState.IDLE,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Tessie Media platform from a config entry."""
+    coordinators = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(TessieMediaEntity(coordinator) for coordinator in coordinators)
+
+
+class TessieMediaEntity(TessieEntity, MediaPlayerEntity):
+    """Vehicle Location Media Class."""
+
+    _attr_name = None
+    _attr_device_class = MediaPlayerDeviceClass.SPEAKER
+
+    def __init__(
+        self,
+        coordinator: TessieDataUpdateCoordinator,
+    ) -> None:
+        """Initialize the media player entity."""
+        super().__init__(coordinator, "media")
+
+    @property
+    def state(self) -> MediaPlayerState | None:
+        """State of the player."""
+        return STATES.get(
+            self.get("vehicle_state_media_info_media_playback_status"),
+            MediaPlayerState.OFF,
+        )
+
+    @property
+    def volume_level(self) -> float | None:
+        """Volume level of the media player (0..1)."""
+        return self.get("vehicle_state_media_info_audio_volume", 0) / self.get(
+            "vehicle_state_media_info_audio_volume_max", 10.333333
+        )
+
+    @property
+    def media_duration(self) -> int | None:
+        """Duration of current playing media in seconds."""
+        duration = self.get("vehicle_state_media_info_now_playing_duration")
+        # Return None if duration is 0
+        return duration / 1000 if duration else None
+
+    @property
+    def media_position(self) -> int | None:
+        """Position of current playing media in seconds."""
+        position = self.get("vehicle_state_media_info_now_playing_elapsed")
+        # Return None if position is 0
+        return position / 1000 if position else None
```

Comment:
```suggestion
        if (position := self.get("vehicle_state_media_info_now_playing_elapsed")) is not None:
            return position / 1000
        return None
```
What do you think about using a walrus? Look better than all the inline ifs

Reply:
Lets see if it breaks my test coverage.

Reply:
I mean it written on my phone 😂